### PR TITLE
Update dataset and spicepod definitions/usage

### PR DESCRIPTION
**Pull request by mitchdevenport**
Updates data and spicepod definitions and all usage to the current definition agreed upon for the M1 demo.

The dataset wizard has been simplified for the time being to accommodate the new `from` field.

### CODE REVIEW

File: bin/spice/pkg/cli/cmd/dataset.go [hunk 1]
```diff
@@ -66,12 +55,13 @@ spice dataset configure
...
-			Source: datasetSource,
+			From: datasetLocation,
+			Name: datasetName,
 			Acceleration: &api.Acceleration{
-				Enabled: accelerateDataset,
-				Refresh: time.Hour,
+				Enabled:         accelerateDataset,
+				RefreshInterval: time.Hour,
+				RefreshMode:     api.REFRESH_MODE_FULL,
+				Retention:       30 * time.Minute,
```
Comment (phillipleblanc) [inline bin/spice/pkg/cli/cmd/dataset.go]:
```suggestion
```

I don't think we should set a retention by default - especially given its a full refresh

File: bin/spice/pkg/cli/cmd/init.go [hunk 2]
```diff
@@ -31,8 +31,10 @@ spice init trader
 			}
 		}
 
-		skeletonPod := &spec.PodSpec{
-			Name: podName,
+		skeletonPod := &api.Pod{
+			Name:    podName,
+			Version: "v1beta2",
```
Comment (phillipleblanc) [inline bin/spice/pkg/cli/cmd/init.go]:
```suggestion
			Version: "v1beta1",
```

Isn't this beta1?

File: bin/spiced/src/lib.rs [hunk 3]
```diff
@@ -65,73 +65,60 @@ pub async fn run(args: Args) -> Result<()> {
...
-                let data_source = Box::leak(data_source);
-
-                df.attach(
-                    &ds.name,
-                    data_source,
-                    databackend::DataBackendType::default(),
+        let mut from_parts: Vec<&str> = ds.from.split('/').collect();
+        let source = from_parts[0];
+        from_parts.remove(0);
+        let dataset_path = from_parts.join(".");
```
Comment (phillipleblanc) [inline bin/spiced/src/lib.rs:71]: The idea is that just the string before the first slash is used to select the data source - everything after will be left as is to be interpreted by the data source

File: bin/spiced/src/lib.rs [hunk 4]
```diff
@@ -65,73 +65,60 @@ pub async fn run(args: Args) -> Result<()> {
...
-                df.attach_backend(&ds.name, databackend::DataBackendType::Memtable)
-                    .context(UnableToAttachDataSourceSnafu {
-                        data_source: "direct",
-                    })?;
+            "dremio" => {
+                let dremio_auth = auth.get(source);
+                Box::new(
+                    datasource::flight::Flight::new(
+                        dremio_auth,
+                        "http://dremio-4mimamg7rdeve.eastus.cloudapp.azure.com:32010".to_string(),
```
Comment (phillipleblanc) [inline bin/spiced/src/lib.rs:92]: This should come from params?

File: bin/spiced/src/lib.rs [hunk 5]
```diff
@@ -65,73 +65,60 @@ pub async fn run(args: Args) -> Result<()> {
...
+            }),
+            _ => UnknownDataSourceSnafu {
+                data_source: source,
+            }
+            .fail()?,
+        };
+
+        let data_source = Box::leak(data_source);
+
+        let fq_dataset_name = format!("{}.{}", dataset_path, ds.name);
```
Comment (phillipleblanc) [inline bin/spiced/src/lib.rs:111]: Let's just attach it as the ds.name - not the fq name. As mentioned in the discussion, the fq name could support more levels than datafusion allows.

File: bin/spiced/src/lib.rs (hunk 3 above)
Comment (mitchdevenport) [reply bin/spiced/src/lib.rs:71]: Yes, this is just temporary alignment of what we have to the dataset spec. Not an overhaul of the OSS internals to the end state.

File: crates/runtime/src/http/v1.rs [hunk 6]
```diff
@@ -23,12 +23,7 @@ pub(crate) mod datasets {
     ) -> Json<Vec<Dataset>> {
         let datasets: Vec<Dataset> = match filter.source {
             Some(source) => match source {
-                DatasetSource::Sink => app
-                    .datasets
-                    .iter()
-                    .filter(|d| d.source.is_none())
-                    .cloned()
-                    .collect(),
+                DatasetSource::Sink => app.datasets.clone(),
```
Comment (phillipleblanc) [inline crates/runtime/src/http/v1.rs:26]: Why did you change this?

Comment (phillipleblanc) [reply crates/runtime/src/http/v1.rs:26]: I'll fix this in a follow up PR

Comment (mitchdevenport) [reply crates/runtime/src/http/v1.rs:26]: source is no longer its own optional field. All datasets implicitly have a source now (everything before the first `/` or `local`)

File: bin/spiced/src/lib.rs (hunk 4 above)
Comment (mitchdevenport) [reply bin/spiced/src/lib.rs:92]: Good point, will make that change

File: crates/spicepod/src/spec.rs [hunk 7]
```diff
@@ -8,6 +8,7 @@ use crate::component::{dataset::Dataset, ComponentOrReference};
 #[serde(rename_all = "lowercase")]
 pub enum SpicepodVersion {
     V1Beta1,
+    V1Beta2,
```
Comment (phillipleblanc) [inline crates/spicepod/src/spec.rs]: Why a new version?

File: bin/spiced/src/lib.rs (hunk 5 above)
Comment (mitchdevenport) [reply bin/spiced/src/lib.rs:111]: If we do that then the dataset names will need to be unique even across sources, so I don't think that's a great solution. I think our best bet is what we do in Ducker (automatically replacing "." with "_") which we've already started doing.

File: crates/spicepod/src/spec.rs (hunk 7 above)
Comment (mitchdevenport) [reply crates/spicepod/src/spec.rs]: I assumed V1Beta1 referred to the very first iteration. Reverted.

File: crates/runtime/src/http/v1.rs (hunk 6 above)
Comment (mitchdevenport) [reply crates/runtime/src/http/v1.rs:26]: So no filtering is needed here

Comment (lukekim) [reply crates/runtime/src/http/v1.rs:26]: On thinking about this, I think we should be even more explicit. If we are generally using domain names like `spice.ai` or `databricks.com` as the "source" then we should use `localhost` as the local source. E.g. `localhost/org/app`

File: bin/spiced/src/lib.rs (hunk 3 above)
Comment (mitchdevenport) [reply bin/spiced/src/lib.rs:71]: Actually, upon further thought I agree with you that we should start doing this now. I'll make this change too.

File: bin/spiced/src/lib.rs (hunk 5 above)
Comment (phillipleblanc) [reply bin/spiced/src/lib.rs:111]: I really dislike that approach - let's find a better way. I'm fine with the constraint that dataset names need to be unique for now.

Comment (phillipleblanc) [reply bin/spiced/src/lib.rs:111]: Replacing dataset names is too much magic - especially since customers will need to know about that to make queries

File: crates/runtime/src/http/v1.rs (hunk 6 above)
Comment (phillipleblanc) [reply crates/runtime/src/http/v1.rs:26]: If we don't filter here - this the query parameter is pointless and doesn't actually do what I need it to :)

Comment (mitchdevenport) [reply crates/runtime/src/http/v1.rs:26]: Ok, will let you fix it up

Comment (phillipleblanc) [reply crates/runtime/src/http/v1.rs:26]: So then this changes to a filter on source - and to get the datasets that are publishable, I filter to the ones with a `localhost` source.
